lib/versions.nix

```diff
@@ -35,15 +40,69 @@ rec {
   */
   patch = v: builtins.elemAt (splitVersion v) 2;
 
+  /* Get string of the first n parts of a version string.
+
+    Example:
+    - truncate 2 "1.2.3-stuff"
+      => "1.2"
+    - truncate 4 "1.2.3-stuff"
+      => "1.2.3.stuff"
```

Comment:
Wouldn't it be?

```suggestion
    - truncate 3 "1.2.3-stuff"
      => "1.2.3-stuff"
```

Reply:
Well, the pitfalls of this truncation function is that it follows the implementation of `builtins.splitVersion`, which has the following behaviour 
```nix
builtins.splitVersion "1.2.3-stuff" == [ "1" "2" "3" "stuff" ]
```
In the same way the current (in nixpkgs master branch) `majorMinor` implementation satisfies the following:
```nix
lib.versions.majorMinor "1.2-stuff" == "1.2"
lib.versions.majorMinor "1-stuff" == "1.stuff"
```
I merely abstracted away the body of the function to generalize it to `majorMinorPatch`.

I agree this is not intuitive, but that's the best I can do that does not break compatibility and extends the current library quite uniformly.

Reply:
I see, maybe add a note on `splitVersion` in the documentation.

Reply:
done

Reply:
Maybe you should also add a note in the documentation of `truncate` that it doesn't preserve the delimiter type explicitly.